### ISSUE TITLE
Why I run SoReg algorithm got an Infinity  problem!

### ISSUE THREAD

**Issue by Jakye** [guoguibing/librec#13]
According to your suggestion ,I set parameter for  SoReg algorithm as fallows:
SoReg.beta=0.001
val.reg.user=0.001
val.reg.item=0.001
in librec.conf
However, when I run the  algorithm in Ciao data set ,I got the problem as fallows:
2015-04-29 14:53:26,384 [INFO] Training: F:\dataset\CiaoDVDs\ratings.txt, ratio: 0.8
2015-04-29 14:53:26,868 [DEBUG] Dataset: {Users, Items, Ratings} = {17615, 16121, 72665}, Scale = {1.0, 2.0, 3.0, 4.0, 5.0}
2015-04-29 14:53:27,040 [DEBUG] training amount: 57929, test amount: 14416
2015-04-29 14:53:27,040 [DEBUG] Social dataset: F:\dataset\CiaoDVDs\trust.txt
2015-04-29 14:53:27,196 [DEBUG] Dataset: {Users, Users, Links} = {19533, 19533, 40133}, Scale = {1.0}
2015-04-29 14:53:27,291 [DEBUG] SoReg: 0.01,-1.0,0.001,0.001,0.001,0.001,10,100,true
2015-04-29 14:53:27,525 [DEBUG] SoReg iter 1: errs = 121772.8, delta_errs = -121772.8, loss = 121965.305, delta_loss = -121965.305, learn_rate = 0.01
2015-04-29 14:53:27,649 [DEBUG] SoReg iter 2: errs = 949440.1, delta_errs = -827667.3, loss = 950113.56, delta_loss = -828148.25, learn_rate = 0.01
2015-04-29 14:53:27,728 [DEBUG] SoReg iter 3: errs = 1.42163802E9, delta_errs = -1.42068851E9, loss = 1.42168947E9, delta_loss = -1.42073933E9, learn_rate = 0.005
2015-04-29 14:53:27,839 [DEBUG] SoReg iter 4: errs = 3.23917665E18, delta_errs = -3.23917665E18, loss = 3.23917665E18, delta_loss = -3.23917665E18, learn_rate = 0.0025
2015-04-29 14:53:27,901 [DEBUG] SoReg iter 5: errs = Infinity, delta_errs = -Infinity, loss = Infinity, delta_loss = -Infinity, learn_rate = 0.00125
2015-04-29 14:53:27,996 [DEBUG] SoReg iter 6: errs = Infinity, delta_errs = -Infinity, loss = Infinity, delta_loss = -Infinity, learn_rate = 6.25E-4
2015-04-29 14:53:28,058 [DEBUG] SoReg iter 7: errs = Infinity, delta_errs = -Infinity, loss = Infinity, delta_loss = -Infinity, learn_rate = 3.125E-4
2015-04-29 14:53:28,058 [ERROR] Loss = NaN or Infinity: current settings cannot train the recommender! Try other settings instead!

Would you like tell me why I got such problem!
